packages/plugin-ext/src/main/browser/languages-main.ts

```diff
@@ -698,35 +693,56 @@ export class LanguagesMainImpl implements LanguagesMain {
     }
 }
 
-function reviveMarker(marker: MarkerData): monaco.editor.IMarkerData {
-    const monacoMarker: monaco.editor.IMarkerData = {
+function reviveMarker(marker: MarkerData): vst.Diagnostic {
+    const monacoMarker: vst.Diagnostic = {
         code: marker.code,
-        severity: marker.severity,
+        severity: reviveSeverity(marker.severity),
+        range: reviveRange(marker.startLineNumber, marker.startColumn, marker.endLineNumber, marker.endColumn),
         message: marker.message,
         source: marker.source,
-        startLineNumber: marker.startLineNumber,
-        startColumn: marker.startColumn,
-        endLineNumber: marker.endLineNumber,
-        endColumn: marker.endColumn,
         relatedInformation: undefined
     };
+
     if (marker.relatedInformation) {
-        monacoMarker.relatedInformation = [];
-        for (const ri of marker.relatedInformation) {
-            monacoMarker.relatedInformation.push({
-                resource: monaco.Uri.revive(ri.resource),
-                message: ri.message,
-                startLineNumber: ri.startLineNumber,
-                startColumn: ri.startColumn,
-                endLineNumber: ri.endLineNumber,
-                endColumn: ri.endColumn
-            });
-        }
+        monacoMarker.relatedInformation = marker.relatedInformation.map(reviveRelated);
     }
 
     return monacoMarker;
 }
 
+function reviveSeverity(severity: MarkerSeverity): vst.DiagnosticSeverity {
+    switch (severity) {
+        case MarkerSeverity.Error: return vst.DiagnosticSeverity.Error;
+        case MarkerSeverity.Warning: return vst.DiagnosticSeverity.Warning;
+        case MarkerSeverity.Info: return vst.DiagnosticSeverity.Information;
+        case MarkerSeverity.Hint: return vst.DiagnosticSeverity.Hint;
+    }
+}
+
+function reviveRange(startLine: number, startColumn: number, endLine: number, endColumn: number): vst.Range {
+    // note: language server range is 0-based, marker is 1-based, so need to deduct 1 here
```

Comment:
is it possible to use that class that has a lot of converters with tests ? https://github.com/theia-ide/theia/blob/1c697d30ade79c79903ca4c45fb352afe965c0c7/packages/plugin-ext/src/plugin/type-converters.ts

there is no existing method yet that covers this convert ?